Contract API Documentation (#5162)
ethereum/web3.js: docs/docs/glossary/json_interface.md

---

> title: Json Interface

Reviewer: ```suggestion
title: JSON Interface
```

---

> The json interface is a json object describing the [Application Binary Interface (ABI)](https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI) for an Ethereum smart contract.

Reviewer: ```suggestion
The JSON interface is a JSON object describing the [Application Binary Interface (ABI)](https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI) for an Ethereum smart contract.
```

> The JSON interface is a JSON object describing the [Application Binary Interface (ABI)](https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI) for an Ethereum smart contract.

Reviewer 2: ```suggestion
The JSON interface is a JSON object describing the [Application Binary Interface (ABI)](https://docs.soliditylang.org/en/develop/abi-spec.html) for an Ethereum smart contract.
```

> Using this json interface web3.js is able to create JavaScript object representing the smart contract and its methods and events using the web3.eth.Contract object.

Reviewer: ```suggestion
Using this JSON interface, web3.js is able to create a JavaScript object representing the smart contract , its methods and events using the web3.eth.Contract object.
```

### Specification

#### Functions

-   `type`: `"function"`, `"constructor"` (can be omitted, defaulting to `"function"`; `"fallback"` also possible but not relevant in web3.js);
-   `name`: the name of the function (only present for function types);
-   `constant`: true if function is specified to not modify the blockchain state;
-   `payable`: true if function accepts ether, defaults to false;

> -   `stateMutability`: a string with one of the following values: pure (specified to not read blockchain state), view (same as constant above), nonpayable and payable (same as payable above);

Reviewer: ```suggestion
-   `stateMutability`: a string with one of the following values: pure (specified to not read blockchain state), view (same as constant above), non-payable and payable (same as payable above);
```

-   `inputs`: an array of objects, each of which contains:
    -- `name`: the name of the parameter;
    -- `type`: the canonical type of the parameter.

> -   `outputs`: an array of objects same as inputs, can be omitted if no outputs exist.

Reviewer: ```suggestion
-   `outputs`: an array of objects, same as inputs, can be omitted if no outputs exist.
```

#### Events

-   `type`: always "event"
-   `name`: the name of the event;
-   `inputs`: an array of objects, each of which contains:
    -- `name`: the name of the parameter;
    -- `type`: the canonical type of the parameter.

>     -- `indexed`: true if the field is part of the log’s topics, false if it one of the log’s data segment.

Reviewer: ```suggestion
    -- `indexed`: true if the field is part of the log’s topics, false if it is one of the log’s data segment.
```
